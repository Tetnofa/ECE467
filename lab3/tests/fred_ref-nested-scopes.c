int a;

void f1() {
  bool a;
  int b;
  b = a;
}
