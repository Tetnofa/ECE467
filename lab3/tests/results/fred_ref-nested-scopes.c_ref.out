Program [useIO=0](1,0)
	Scalar Declaration	Int(1,0)
		Identifier[name:a](1,4)
	Function[isProto=0]	Void(3,0)
		Identifier[name:f1](3,5)
		Scope(3,10)
			Scalar Declaration			Bool(4,2)
				Identifier[name:a](4,7)
			Scalar Declaration			Int(5,2)
				Identifier[name:b](5,6)
			Assignment(6,2)
				Reference(6,2)
					Identifier[name:b](6,2)
				Int Expression(6,6)
					Reference(6,6)
						Identifier[name:a](6,6)
sema: 6:2 : type mismatch 
